File: ulohy/08-bod-v-kruhu/reseni/main.cs
using System;

internal class Program
{
    static void Main(string[] args)
    {
        double diameter = ReadDouble();
        double centerX = ReadDouble();
        double centerY = ReadDouble();
        double pointX = ReadDouble();
        double pointY = ReadDouble();

        bool liesInside = LiesInside(diameter / 2, centerX, centerY, pointX, pointY);

        Console.WriteLine(liesInside ? "inside" : "outside");
    }

    static bool LiesInside(double radius, double centerX, double centerY, double pointX, double pointY)
    {
        double a = Math.Abs(centerX - pointX);
        double b = Math.Abs(centerY - pointY);
        double c = Math.Sqrt(Math.Pow(a, 2) + Math.Pow(b, 2));

        return c <= radius;
    }

    static double ReadDouble()
    {
        return double.Parse(Console.ReadLine());
    }
}

File: ulohy/08-bod-v-kruhu/reseni/main.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        double diameter = scanner.nextDouble();
        double centerX = scanner.nextDouble();
        double centerY = scanner.nextDouble();
        double pointX = scanner.nextDouble();
        double pointY = scanner.nextDouble();

        boolean liesInside = liesInside(diameter / 2, centerX, centerY, pointX, pointY);

        System.out.println(liesInside ? "inside" : "outside");
    }

    static boolean liesInside(double radius, double centerX, double centerY, double pointX, double pointY) {
        double a = Math.abs(centerX - pointX);
        double b = Math.abs(centerY - pointY);
        double c = Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2));

        return c <= radius;
    }
}
